mlflow/utils/autologging_utils/client.py

```diff
@@ -0,0 +1,370 @@
+        # is not logged if an exception preempts the call to `flush()`
+        if exc is None and exc_type is None and traceback is None:
+            self.flush(synchronous=True)
+        else:
+            _logger.debug(
+                "Skipping run content logging upon MlflowAutologgingQueueingClient context because"
+                " an exception was raised within the context: %s",
+                exc,
+            )
+
+    def create_run(
+        self,
+        experiment_id: str,
+        start_time: Optional[int] = None,
+        tags: Optional[Dict[str, Any]] = None,
+    ) -> PendingRunId:
+        """
+        Enqueues a CreateRun operation with the specified attributes, returning a `PendingRunId`
+        instance that can be used as input to other client logging APIs (e.g. `log_metrics`,
+        `log_params`, ...).
+
+        :return: A `PendingRunId` that can be passed as the `run_id` parameter to other client
+                 logging APIs, such as `log_params` and `log_metrics`.
+        """
+        tags = tags or {}
+        tags = _truncate_dict(
+            tags, max_key_length=MAX_ENTITY_KEY_LENGTH, max_value_length=MAX_TAG_VAL_LENGTH
+        )
+        run_id = PendingRunId()
+        self._get_pending_operations(run_id).enqueue(
+            create_run=_PendingCreateRun(
+                experiment_id=experiment_id,
+                start_time=start_time,
+                tags=[RunTag(key, str(value)) for key, value in tags.items()],
+            )
+        )
+        return run_id
+
+    def set_terminated(
+        self,
+        run_id: Union[str, PendingRunId],
+        status: Optional[str] = None,
+        end_time: Optional[int] = None,
+    ) -> None:
+        """
+        Enqueues an UpdateRun operation with the specified `status` and `end_time` attributes
+        for the specified `run_id`.
+        """
+        self._get_pending_operations(run_id).enqueue(
+            set_terminated=_PendingSetTerminated(status=status, end_time=end_time,)
+        )
+
+    def log_params(self, run_id: Union[str, PendingRunId], params: Dict[str, Any]) -> None:
+        """
+        Enqueues a collection of Parameters to be logged to the run specified by `run_id`.
+        """
+        params = _truncate_dict(
+            params, max_key_length=MAX_ENTITY_KEY_LENGTH, max_value_length=MAX_PARAM_VAL_LENGTH
+        )
+        params_arr = [Param(key, str(value)) for key, value in params.items()]
+        self._get_pending_operations(run_id).enqueue(params=params_arr)
+
+    def log_metrics(
+        self,
+        run_id: Union[str, PendingRunId],
+        metrics: Dict[str, float],
+        step: Optional[int] = None,
+    ) -> None:
+        """
+        Enqueues a collection of Metrics to be logged to the run specified by `run_id` at the
+        step specified by `step`.
+        """
+        metrics = _truncate_dict(metrics, max_key_length=MAX_ENTITY_KEY_LENGTH)
+        timestamp = int(time.time() * 1000)
```

Comment:
timestamp_ms to include the unit for better readability.

Reply:
Changed to `timestamp_ms` - thanks for the suggestion!